src/librustdoc/html/render/search_index.rs

```diff
@@ -241,6 +241,26 @@ impl SerializedSearchIndex {
         self.alias_pointers.push(alias_pointer);
         index
     }
+    fn add_entry(&mut self, name: Symbol, entry_data: EntryData, desc: String) -> usize {
```

Comment:
Might be worth adding a code comment (or a doc comment on the method) to explain a bit what's being done in here.

Reply:
Okay, I've added a comment.